pkg/operation/seed/scheduler/gardener_kube_scheduler_test.go

```diff
@@ -0,0 +1,440 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scheduler_test
+
+import (
+	"context"
+
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	. "github.com/gardener/gardener/pkg/operation/seed/scheduler"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+
+	"github.com/Masterminds/semver"
+	resourcesv1alpha1 "github.com/gardener/gardener-resource-manager/pkg/apis/resources/v1alpha1"
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Scheduler", func() {
+	const (
+		deployNS = "test-namespace"
+		image    = "foo:v1"
+	)
+
+	var (
+		ctx                                              context.Context
+		c                                                client.Client
+		sched                                            component.DeployWaiter
+		version                                          *semver.Version
+		webhookClientConfig, expectedWebhookClientConfig *admissionregistrationv1beta1.WebhookClientConfig
+		expectedLabels                                   map[string]string
+		codec                                            serializer.CodecFactory
+		assertScheduler                                  = func() {
+			Context("New succeeds", func() {
+				var managedResourceSecret *corev1.Secret
+
+				BeforeEach(func() {
+					managedResourceSecret = &corev1.Secret{}
+				})
+
+				JustBeforeEach(func() {
+					s, err := New(c, deployNS, image, version, &dummyConfigurator{}, webhookClientConfig)
+					Expect(err).NotTo(HaveOccurred(), "New succeeds")
+
+					sched = s
+				})
+
+				Context("Deploy", func() {
+					JustBeforeEach(func() {
+						Expect(sched.Deploy(ctx)).NotTo(HaveOccurred(), "deploy succeeds")
```

Comment:
```suggestion
						Expect(sched.Deploy(ctx)).To(Suceed(), "deploy succeeds")
```

you could use the same for other occurrences